File: src/Repository/RepositoryFramework.Abstractions/Consumers/Interfaces/ICommand.cs
namespace RepositoryFramework
{
    /// <summary>
    /// Interface for your CQRS pattern, with Insert, Update and Delete methods.
    /// This is the interface injected by the framework and that you may use for your purpose.
    /// </summary>
    /// <typeparam name="T">Model used for your repository.</typeparam>
    /// <typeparam name="TKey">Key to insert, update or delete your data from repository.</typeparam>
    public interface ICommand<T, TKey> : ICommandPattern<T, TKey>
        where TKey : notnull
    {
    }
}

File: src/Repository/repositoryframework.api.client.typescript/src/rystem/src/interfaces/ICommand.ts
import { BatchBuilder } from "../engine/batch/BatchBuilder";
import { State } from "../models/State";

export interface ICommand<T, TKey> {
    insert(key: TKey, value: T): Promise<State<T, TKey>>;
    update(key: TKey, value: T): Promise<State<T, TKey>>;
    delete(key: TKey): Promise<State<T, TKey>>;
    batch(): BatchBuilder<T, TKey>;
}

export interface ICommandPattern {
    insert(key: any, value: any): Promise<State<any, any>>;
    update(key: any, value: any): Promise<State<any, any>>;
    delete(key: any): Promise<State<any, any>>;
    batch(): BatchBuilder<any, any>;
}
